### PR TITLE
Fix jsonschema deep property

### PR DESCRIPTION
**Pull request by Romakita**
## Informations

Type | Status | Migration
---|---|---
Fix | Ready | Yes/No

****

## Description
Related issues: #210, #198 (error message format).

Thanks @lwallent  for your example the issue ;)

### CODE REVIEW

File: src/ajv/interfaces/IAjvSettings.ts [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
-
 declare interface IServerSettings {
     ajv: IAjvSettings;
 }
 
-export type ErrorFormatter = (error: any) => string;
+/**
+ *
+ */
+export interface AjvErrorObject extends ErrorObject {
```
Comment (lwallent) [inline src/ajv/interfaces/IAjvSettings.ts]: 👍  Thumbs up for the added type. Sorry, for being to lazy to do this :)

File: src/ajv/services/AjvService.ts [hunk 2]
```diff
@@ -19,29 +20,61 @@ export class AjvService extends ValidationService {
...
+     * @param error
+     * @returns {string}
+     */
+    private defaultFormatter(error: AjvErrorObject) {
+        let value = "";
+        if (this.options.verbose) {
+            value = `, value "${error.data}"`;
+        }
+        return `At ${error.modelName}${error.dataPath}${value} ${error.message}`;
+    }
```
Comment (lwallent) [inline src/ajv/services/AjvService.ts]: Thanks for cleaning up my sloppy code 👍

File: src/jsonschema/registries/JsonSchemesRegistry.ts [hunk 3]
```diff
@@ -112,23 +112,23 @@ export class JsonSchemaRegistry extends Registry<any, Partial<JsonSchema>> {
...
 
-                schema[key] = Object.keys(schema[key]).reduce((acc: any, index: any) => {
-                    schema[key][index] = this.findReferences(schema[key][index], definitions);
-                    return schema[key];
-                }, obj);
+                Object
+                    .keys(value)
+                    .forEach((index: any) => {
+                        this.findReferences(value[index], definitions);
+                    });
```
Comment (lwallent) [inline src/jsonschema/registries/JsonSchemesRegistry.ts]: Thanks for cleaning up your own code as well :)

File: test/units/ajv/services/AjvService.spec.ts [hunk 4]
```diff
@@ -2,9 +2,8 @@ import "../../../../src/ajv";
 import {AjvService} from "../../../../src/ajv/services/AjvService";
 import {globalServerSettings} from "../../../../src/config";
 import {JsonSchemesService} from "../../../../src/jsonschema";
-import {ParseExpressionError} from "../../../../src/mvc/errors/ParseExpressionError";
 import {inject} from "../../../../src/testing";
-import {JsonFoo, JsonFoo2} from "../../../helper/classes";
+import {JsonFoo, JsonFoo2, Nested, Stuff, Thingy} from "../../../helper/classes";
```
Comment (lwallent) [inline test/units/ajv/services/AjvService.spec.ts]: OMG, I have just messed up the code once again without even doing a PR :-)

File: src/jsonschema/registries/JsonSchemesRegistry.ts (hunk 3 above)
Comment (Romakita) [reply src/jsonschema/registries/JsonSchemesRegistry.ts]: Well, I don't understand why I written this shit code XD. Thanks again for your review ;)

**Pull request merged**